portlet-parent/wicketstuff-portlet/src/main/java/org/apache/wicket/portlet/WicketPortlet.java

```diff
@@ -412,23 +412,7 @@ private void processMimeResponseRequest(String wicketURL, final PortletRequest r
 				// processing the request
 
 				String redirectLocation = responseState.getRedirectLocation();
-				String ajaxRedirectLocation = responseState.getAjaxRedirectLocation();
-				if (ajaxRedirectLocation != null) {
-					// Ajax redirect
-					responseState.clear();
-					responseState.setDateHeader("Date", System.currentTimeMillis());
-					responseState.setDateHeader("Expires", 0);
-					responseState.setHeader("Pragma", "no-cache");
-					responseState.setHeader("Cache-Control", "no-cache, no-store");
-					//client side javascript needs the Ajax-Location header see wicket-ajax-jquery.js line 771
-					responseState.setHeader("Ajax-Location", ajaxRedirectLocation);//

```

Comment:
Client side javascript needs the Ajax-Location header in order to redirect. Please  see [wicket-ajax-jquery.js](https://github.com/wicketstuff/core/blob/master/portlet-parent/wicketstuff-portlet/src/main/java/org/apache/wicket/portlet/ajax/res/js/wicket-portlet-ajax.js) line 14.

Reply:
Also look at my very first [commit](https://github.com/wicketstuff/core/pull/476/commits/7831805b42cbb30edd33ce2f1cce1ca9f8f829df) where I fixed the redirect after ajax request behavior.